docs/DeveloperGuide_Winery.md

```diff
@@ -0,0 +1,622 @@
+  * [Project org.eclipse.winery.model.selfservice](#project-org.eclipse.winery.model.csar.toscametafile)  
+  * [Project org.eclipse.winery.model.tosca](#project-org.eclipse.winery.model.csar.toscametafile)  
+  * [Project org.eclipse.winery.repository.client](#project-org.eclipse.winery.repository.client)  
+  * [Project org.eclipse.winery.repository](#project-org.eclipse.winery.repository)  
+  * [Project org.eclipse.winery.topologymodeler](#project-org.eclipse.winery.topologymodeler)  
+- [Winery's ID-System](#winery's-id-system)  
+  * [AdminId](#adminid)  
+  * [ToscaComponentId](#toscacomponentid)  
+- [Filesystem Layout](#filesystem-layout)
+- [Rest Resources](#rest-resources)  
+  * [URL Schema](#url-schema)    
+  * [Collections of Components](#collections-of-components)  
+  * [Component Instances](#component-instances)  
+    + [AbstractComponentInstanceResourceWithNameDerivedFromAbstractFinal](#abstractcomponentinstanceresourcewithnamederivedfromabstractfinal)  
+	+ [AbstractComponentInstanceWithReferencesResource](#abstractcomponentinstancewithreferencesresource)  
+	+ [GenericImportResource](#genericimportresource)  
+	+ [PolicyTemplateResource](#policytemplateresource)  
+- [Working with the Repository](#working-with-the-repository)  
+  * [IWineryRepositoryCommon](#iwineryrepositorycommon)  
+  * [IGenericRepository](#igenericrepository)  
+	+ [IRepository](#irepository)  
+	+ [IWineryRepository](#iwineryrepository)  
+- [Shared JSPs and TAGs](#shared-jsps-and-tags)  
+- [Type, Template and Artifact Management](#type-template-and-artifact-management)
+- [Topology Modeler](#topology-modeler)
+- [Debugging Hints](#debugging-hints)  
+  * [Debugging JavaScriptCode](#debugging-javascriptcode)	 
+    + [Chrome](#chrome)  
+	+ [Firefox](#firefox)
+  * [Automatic Browser Refresh](#Automatic-browser-refresh)  
+  * [Faster Redeployment](#faster-redeployment)
+- [Miscellaneous Hints](#miscellaneous-hints)  
+  * [Generating the Right Output](#generating-the-right-output)  
+  * [Other Troubleshootings](#other-troubleshootings)   
+  * ["name" vs. "id" at Entities](#name-vs-id-at-entities)  
+  * [Possible Attachments of Artifacts](#possible-attachments-of-artifacts)
+- [Example Repository](#example-repository)
+- [Abbreviations](#abbreviations)
+- [References](#references)
+- [List of Figures](#list-of-figures)
+
+<!-- tocstop -->
+
+---
+
+#### Overview
+
+This document provides the design ideas of Winery.
+
+Winery is a Web based environment ro graphically model TOSCA topologies and plans managing these
+topologies. It is an Eclipse project and thus support is available through its project page
+[https://projects.eclipse.org/projects/soa.winery] (https://projects.eclipse.org/projects/soa.winery). Winery is also part of the OpenTOSCA ecosystem where more
+information is available at [http://www.opentosca.org](http://www.opentosca.org).
```

Comment:
Hyperlinks are written <http://example.com/like-this> in markdown. In GitHub, you can use the plain link. No need to bouble encode.